### PR TITLE
Add chrome-api modules for code review

### PR DESCRIPTION
**Pull request by srsudar**

### CODE REVIEW

File: chromeapp/app/scripts/chrome-apis/file-system.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+exports.isWritableEntry = function(entry) {
+  return new Promise(function(resolve) {
+    chrome.fileSystem.isWritableEntry(entry, function(isWritable) {
+      resolve(isWritable);
+    });
+  });
+};
+
+/**
+ * The original Chrome callback takes two parameters: an entry and an array of
```
Comment (mdwelsh) [inline chromeapp/app/scripts/chrome-apis/file-system.js:50]: Can you file a crbug about this? It does not make sense that the documentation doesn't explain what the second parameter is for (and I agree that the docs are confusing).

**Comment by mdwelsh** [comment]
Are there tests for these modules?

File: chromeapp/app/scripts/chrome-apis/file-system.js (hunk 1 above)
Comment (srsudar) [inline chromeapp/app/scripts/chrome-apis/file-system.js:50]: [Done](https://bugs.chromium.org/p/chromium/issues/detail?id=638782).

**Comment by srsudar** [comment]
No, I didn't write tests for these modules. I created them was to make them easily mock-able for testing and to convert them to use Promises rather than callbacks, which I find easier to use. I'd considered it wrapper-ish enough to not include tests. Should I add some?

Chrome makes its APIs available via the global `chrome` variable. Rather than trying to mock this global, if I add tests I'll likely add an additional module that just returns `chrome.sockets.udp` or whatever API I'm wrapping and test it invokes things on that. Certainly doable.

**Pull request merged**